src/utils/usePromiseResolver.test.tsx

```diff
@@ -0,0 +1,74 @@
+import {FunctionComponent} from 'react';
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import {mount, ReactWrapper} from 'enzyme';
+import {usePromiseResolver} from './usePromiseResolver';
+import { mockStripe } from '../../test/mocks';
+
+const TestComponentInner: FunctionComponent<{value: any}> = () => null
+const TestComponent: FunctionComponent<{promiseLike: any}> = ({promiseLike}) => {
+  const value = usePromiseResolver(promiseLike);
+  return <TestComponentInner value={value} />
+};
+
+const getHookValue = (wrapper: ReactWrapper) => wrapper.find(TestComponentInner).prop('value')
+
+const createController = (): [Promise<unknown>, (value?: unknown) => Promise<void>, (reason?: any) => Promise<void>] => {
+  let resolveFn: (value?: unknown) => Promise<void> = () => Promise.resolve()
+  let rejectFn: (reason?: any) => Promise<void> = () => Promise.resolve()
+
+  const promise = new Promise((resolve, reject) => {
+    const createVoidPromise = () => promise.then(() => undefined, () => undefined)
+
+    resolveFn = (value) => {
+      resolve(value)
+      return createVoidPromise()
+    }
+
+    rejectFn = (reason) => {
+      reject(reason)
+      return createVoidPromise()
+    }
+  })
+
+  return [promise, resolveFn, rejectFn]
+}
+
+describe('usePromiseResolver', () => {
+  let stripe: ReturnType<typeof mockStripe>;
+
+  beforeEach(() => {
+    stripe = mockStripe();
+  })
+
+  it('returns value on mount when not promise given', () => {
+    const wrapper = mount(<TestComponent promiseLike={stripe} />);
+    expect(wrapper.find(TestComponentInner).prop('value')).toBe(stripe)
```

Comment:
nit: I think we can use the same `getHookValue` helper here:

```suggestion
    expect(getHookValue(wrapper)).toBe(stripe)
```